sound/soc/soc-topology.c

```diff
@@ -2577,6 +2594,9 @@ int snd_soc_tplg_component_remove(struct snd_soc_component *comp, u32 index)
 			case SND_SOC_DOBJ_DAI_LINK:
 				remove_link(comp, dobj, pass);
 				break;
+			case SND_SOC_DOBJ_NONE:
+				unload_link(comp, dobj, pass);
+				break;
```

Comment:
I find the code complicated and confusing:
for a DAI_LINK you use remove_link() but for an DOBJ_NONE you use unload_link() (with the comment that it "unloads dai link".
in addition remove_link() calls snd_soc_remove_dai_link(comp->card, link) so you are going to lose reviewers here.

Reply:
@plbossart For DAI_LINK, which are FE links, are created in soc_tplg_fe_link_create().  So we need to remove it by remove_link(). But for DOBJ_NONE, which are BE links, are not created by topology and they will be removed by [soc_remove_dai_links](https://elixir.bootlin.com/linux/latest/source/sound/soc/soc-core.c#L1001). But It is load by ```snd_soc_tplg_component_load()->soc_tplg_load()->soc_tplg_process_headers()->soc_tplg_load_header()->soc_tplg_link_elems_load()->soc_tplg_link_config()```. So I think it is better to unload it in soc-topology.